packages/react/src/components/TableCard/TableCard.jsx

```diff
@@ -200,18 +200,17 @@ const TableCard = ({
     id,
   ]);
 
-  const renderActionCell = (cellItem) => {
-    const actionList = JSON.parse(cellItem.value);
+  const renderActionCell = ({ value: actionList, rowId }) => {
     return actionList && actionList.length === 1 ? (
       React.createElement(
-        typeof actionList[0].icon === 'string' ? icons[actionList[0].icon] : actionList.icon,
+        typeof actionList[0].icon === 'string' ? icons[actionList[0].icon] : actionList[0].icon,
```

Comment:
The icon list is always an array, even with only one element, so I assume this was meant to work originally or did at some point. Getting coverage over this part is what opened up the whole can of worms, but seemingly it was a simple issue to fix.